packages/svelte/src/internal/client/dom/elements/attributes.js

```diff
@@ -347,16 +347,31 @@ export function set_attributes(
 		} else if (key === '__value' || (key === 'value' && value != null)) {
 			// @ts-ignore
 			element.value = element[key] = element.__value = value;
+		} else if (key === 'selected' && is_option_element) {
+			set_selected(/** @type {HTMLOptionElement} */ (element), value);
 		} else {
 			var name = key;
 			if (!preserve_attribute_case) {
 				name = normalize_attribute(name);
 			}
+			let is_default_value_or_checked = name === 'defaultValue' || name === 'defaultChecked';
 
-			if (value == null && !is_custom_element) {
+			if (value == null && !is_custom_element && !is_default_value_or_checked) {
 				attributes[key] = null;
+				// if we remove the value/checked attributes this also for some reasons reset
+				// the default value so we need to keep track of it and reassign it after the remove
```

Comment:
Ah I see, yeah that's why - now I'm wondering if we should model this differently: Check if the dom element is an input or textarea, and if so, if we come across a `value` or `checked` attribute, skip to the setters. Not sure which solution ends up being less code.